### DISCUSSION TITLE
@DocumentID when receiving data from source that isn't Firestore

### DISCUSSION THREAD

**Discussion by Alex-Coetzee** [firebase/firebase-ios-sdk#11444] [General]
Lets say we have a Struct 

```
struct Person : Identifiable, Codable {

     @DocumentID var id : String?

     var first_name : String?

}
```

In most cases I am querying Firestore directly making the @DocumentID property wrapper very useful.

However, when utilising the newer Functions API I sometimes receive data from the server, which of course does not include a Document ID 

I have simplified the code example to make it easier to follow so let's assume I am sending the COdable as a Person and receiving a Person type back from the function, the issue is that encoding fails due to the @DocumentID wrapper.

`
   let response = try await Functions.functions().httpsCallable("savePerson", requestAs: Person.self, responseAs: Person.self)
`

Is there a way to specificy to ignore the DocumentID wrapper in this scenario when receiving the data back from the server?

Is there a way to customise it using the coding keys or a custom decoder?

Please could you point me in the right direction as it would be much appreciated.

**Comment by nidegen** [comment]
You will probably need to split the struct up into two separate types:
```

struct CorePerson: Codable {
  let firstName = "Peter"
}

struct Person: CorePerson, Identifiable, Codable {
  @DocumentID var id : String
}
```

**Reply by mortenbekditlevsen** [reply]
You can't inherit structs. 😊

**Reply by nidegen** [reply]
😅
